aeron-client/src/main/cpp_wrapper/Subscription.h

```diff
@@ -168,8 +168,59 @@ class Subscription
      */
     std::vector<std::string> localSocketAddresses() const
     {
-        return aeron::concurrent::status::LocalSocketAddressStatus::findAddresses(
-            m_countersReader, channelStatus(), channelStatusId());
+        const int initialVectorSize = 16;
+        uint8_t buffers[initialVectorSize][AERON_CLIENT_MAX_LOCAL_ADDRESS_STR_LEN];
+        aeron_iovec_t initialIovecs[initialVectorSize];
+        std::unique_ptr<uint8_t[]> overflowBuffers;
+        std::unique_ptr<aeron_iovec_t[]> overflowIovecs;
+        aeron_iovec_t *iovecs;
```

Comment:
Why not declare later and assign at the same time? We are not doing K&R C anymore ;-) Less error prone and less semantic distance.

Reply:
The pointer `iovecs` potentially references the data in `overflowIovecs` and `overflowBuffers`, so they need to be at the same scope otherwise the memory would be freed to early.

The `unique_ptr`s are largely there to manage the memory as there is an exception path out of that function and it seemed cleaner to use RAII to free up the memory if it happened to be allocated.